src/javascript/SelectorTypes/MultipleLeftRightSelector/MultipleLeftRightSelectorDnd.jsx

```diff
@@ -0,0 +1,201 @@
+                arrayValue.splice(dnd.current.dragging.originalIndex, 0, current);
+                onChange([...arrayValue]);
+            }
+
+            setDraggedId(null);
+            dnd.current.dragging = null;
+        }
+    }), [arrayValue, onChange]);
+
+    // Right list item drag props
+    const rightListListItemProps = useCallback(value => ({
+        onDragOver: e => {
+            e.preventDefault();
+            e.dataTransfer.dropEffect = 'move';
+
+            // Perform move of the item within the list
+            if (e.dataTransfer.types.includes(DATA_TYPES.MLRS_DRAG_TO_REORDER.toLowerCase())) {
+                if (dnd.current.dragging && dnd.current.dragging.index !== value.index) {
+                    const rect = e.currentTarget.getBoundingClientRect();
+                    const clientOffset = {x: e.clientX, y: e.clientY};
+                    const targetMidPointY = rect.y + (rect.height / 2);
+
+                    // Avoid triggering change for adjacent target
+                    if (clientOffset.y < targetMidPointY && value.index > dnd.current.dragging.index) {
+                        return;
+                    }
+
+                    // Avoid triggering change for adjacent target
+                    if (clientOffset.y > targetMidPointY && value.index < dnd.current.dragging.index) {
+                        return;
+                    }
+
+                    const m = arrayValue[value.index];
+
+                    if (!dnd.current.dragging.originalIndex) {
+                        dnd.current.dragging.originalIndex = dnd.current.dragging.index;
+                    }
+
+                    arrayValue[value.index] = arrayValue[dnd.current.dragging.index];
+                    arrayValue[dnd.current.dragging.index] = m;
+                    dnd.current.dragging.index = value.index;
+                    onChange([...arrayValue]);
+                }
+            }
+
+            // Handle repositioning when moving from another list
+            if (e.dataTransfer.types.includes(DATA_TYPES.MLRS_DRAG_TO_MOVE.toLowerCase())) {
+                if (moved && value.value !== FAKE_VALUE) {
+                    const rect = e.currentTarget.getBoundingClientRect();
+                    const clientOffset = {x: e.clientX, y: e.clientY};
+                    const targetMidPointY = rect.y + (rect.height / 2);
+                    // Avoid triggering change for adjacent target
+                    if (clientOffset.y < targetMidPointY && value.index > dnd.current.dragging.index) {
+                        return;
+                    }
+
+                    // Avoid triggering change for adjacent target
+                    if (clientOffset.y > targetMidPointY && value.index < dnd.current.dragging.index) {
+                        return;
+                    }
+
+                    dnd.current.dragging.index = value.index;
+                    setMoved({
```

Comment:
should be a function to properly capture previous state: `setMoved(moved => ({...moved, index:value.index}))`

Reply:
will change to fcn